File: src/control/ImportController.java
package control;


import model.Route;

import java.time.LocalDate;
import java.util.List;
import java.util.Vector;

/**
 * ImportController
 * Handles all functionality for the use-case import.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class ImportController {

    private RouteController routeController;
    private static ImportController instance;

    /**
     * Private constructor for singleton.
     */
    private ImportController() {
        routeController = RouteController.getInstance();
    }

    /**
     * Singleton method for class.
     *
     * @return instance of class.
     */
    public static ImportController getInstance() {
        if (instance == null) {
            instance = new ImportController();
        }

        return instance;
    }
    
    /**
     * Imports all routes from database.
     * @param rowData
     */
    public void importRoutes(Vector<Vector> rowData) {
        routeController.importRoutes(LocalDate.now());

        List<Route> routes = routeController.getRoutes();

        routes.forEach(route -> {
            Vector row = new Vector();
            row.addElement(Long.valueOf(route.getID()).toString());
            row.addElement(Long.valueOf(route.getDefaultRoute().getID()).toString());
            row.addElement(Integer.valueOf(route.getStops().size()));
            rowData.add(row);
        });
    }
    
}

File: Control/ImportController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;

namespace Control
{
    public class ImportController
    {
        public RouteController RouteCtr { get; private set; }
        private static ImportController instance;

        /// <summary>
        /// Private singleton constructor.
        /// </summary>
        private ImportController() 
        {
            RouteCtr = RouteController.Instance;
        }

        /// <summary>
        /// Singleton method. Returns the instance of the class.
        /// </summary>
        /// <returns>Instance of class.</returns>
        public static ImportController Instance { 
                get { 
                    if (instance == null)
                        instance = new ImportController();
                    return instance;
                }
            }
    
        /// <summary>
        /// Imports all routes from database.
        /// </summary>
        public void ImportRoutes()
        {
            RouteCtr.ImportRoutes(DateTime.Now);
        }

        /// <summary>
        /// Imports all default routes from a .csv file.
        /// </summary>
        /// <param name="pathRoutes">Path of .csv file to be imported.</param>
        /// <param name="pathStops">Path of .csv file to be imported.</param>
        /// <param name="pathCustomers">Path of .csv file to be imported.</param>
        public void ImportFromFile(string pathRoutes, string pathStops, string pathCustomers)
        {
            RouteCtr.ImportRoutesFromFile(DateTime.Now, pathRoutes, pathStops, pathCustomers);
        }
    }
}
